scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+
+
+def _validate_lower_upper(lower, upper):
+    if np.shape(lower) != np.shape(upper):
+        raise ValueError('lower and upper must have the same shape.')
+    if np.isnan(lower).any() or np.isnan(upper).any():
+        raise ValueError('lower and upper must not contain nan.')
+    if (np.isinf(lower) & np.isinf(upper)).any():
+        raise ValueError('lower and upper must not both be infinity.')
+    if (lower > upper).any():
+        raise ValueError('Elements of lower must not be greater than the '
+                         'corresponding elements of upper.')
+
+
+def _validate_x_censored(x, censored):
+    x = np.asarray(x)
+    if x.ndim != 1:
+        raise ValueError('x must be one-dimensinal.')
+    censored = np.asarray(censored)
+    if censored.ndim != 1:
+        raise ValueError('censored must be one-dimensional')
+    if (~np.isfinite(x)).any():
+        raise ValueError('x must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('x and censored must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for MLE fitting.
+
+    Left-, right-, and interval-censored data can be represented.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the ``CensoredData``
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
```

Comment:
(This might have been resolved in our call...)  Yes.  For example, `CensoredData([0, 5, 10], [1, 8, inf])` represents three "observations".  For the first, the true value is in the interval [0, 1], the second true value is in [5, 8], and the third is greater than or equation to 10.

Reply:
Yes, I totally understand now, and I agree that this is a good way for the initializer of `CensoredData` to work. Please do include an example in the class documentation, though.

Reply:
I added examples to the `CensoredData` class docstring.